### PR TITLE
Create ci.yml

### PR DESCRIPTION
**Pull request by taylorpapke**
created ci.yml and added lint and unit tests for the CI pipeline

### CODE REVIEW

**Review by theGaryLarson — APPROVED** [review]
Well-structured and easy to understand. Good job Taylor!

### Some considerations:
**Job Dependency**: Your current setup has the unit tests depending on linting. While this approach helps to save resources and enforces code quality, it could potentially slow down the feedback loop for developers. You might consider running these jobs in parallel to provide quicker feedback

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+              run: ./gradlew lint
+
+            - name: Upload html test report
+              uses: actions/upload-artifact@v2
+              with:
+                name: index.html
+                path: app/build/reports/lint-results-debug.html
+
+  unit-test:
+    needs: [lint]
```
[the remark this answers is not in the document]
Comment (brinkbrink) [reply .github/workflows/ci.yml:27]: Like you said, this is more of a team preference than necessary change. Maybe we should bring it up in team stand-up and do a quick vote? Or maybe our lead engineer has a perspective on this we haven't considered.

[the remark this answers is not in the document]
Comment (theGaryLarson) [reply .github/workflows/ci.yml:27]: Definitely worth having a convo about during stand-up

[the remark this answers is not in the document]
Comment (bcko) [reply .github/workflows/ci.yml:27]: In the workplace, CI/CD processes typically incur charges based on minutes. As Gary mentioned, many organizations execute tests concurrently, but only once specific conditions are fulfilled, such as linting in our scenario. It is generally preferable for more expensive tests to run only after simpler tests like linting have successfully completed, as linting requires less time to execute.

**Review by bcko — APPROVED** [review]

**Review by brinkbrink — APPROVED** [review]

**Pull request merged**